man/machinectl.xml

```diff
@@ -428,6 +428,9 @@
         user may be selected. Use <option>--setenv=</option> to set
         environment variables for the executed process.</para>
 
+        <para>Note that <command>machinectl shell</command> does not propagate the exit code/status of the invoked
+        shell process. Use <command>systemd-run</command> instead if that information is required (see below).</para>
```

Comment:
i didn't want to add a man ref here, as the main comment about systemd-run follows further down, and we point the reader to that using the "(see below)".